datadog/dogstatsd/base.py

```diff
@@ -314,7 +314,10 @@ def close_socket(self):
         Closes connected socket if connected.
         """
         if self.socket:
-            self.socket.close()
+            try:
+                self.socket.close()
+            except Exception as e:
```

Comment:
Since we want to catch that specific `OSError` exception, wouldn't it make more sense to do:

```suggestion
            except OSError as e:
```

Reply:
Sure, I assumed we'd want to silence any exception but since this is only meant to raise `OSError` instances it's likely more correct.